### PR TITLE
Sr persistance layer 2

### PR DESCRIPTION
**Pull request by simen-rekkedal**
The completion of the Persistance Layer

## Description

A continuation on the Persistance_Layer_1 PR
Added SystemRegisterController

## Related Issue(s)
- #369 
- #379 
- #346 
- #361 

## Verification
- [x] **Your** code builds clean without any errors or warnings
- [x] Manual testing done (required)
- [x] Relevant automated test added (if you find this hard, leave it and we'll help out)
- [x] All tests run green

## Documentation
- [ ] User documentation is updated with a separate linked PR in [altinn-studio-docs.](https://github.com/Altinn/altinn-studio-docs) (if applicable)

### CODE REVIEW

File: src/Core/Models/DefaultRights.cs [hunk 1]
```diff
@@ -0,0 +1,8 @@
+﻿namespace Altinn.Platform.Authentication.Core.Models
+{
+    public class DefaultRights
```
Comment (acn-dgopa) [inline src/Core/Models/DefaultRights.cs:3]: I think that we can introduce some namesapce/folder structure that can differntiate this "Default Rights" from the generic rights f.ex Altinn.Platform.Authentication.Core.SystemRegister.Models

File: src/Authentication/Controllers/SystemRegisterController.cs [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+using Altinn.Platform.Authentication.Core.Models;
+using Altinn.Platform.Authentication.Services.Interfaces;
+using Microsoft.AspNetCore.Mvc;
+
+namespace Altinn.Authentication.Controllers;
+
+/// <summary>
+/// CRUD API for SystemRegister
+/// </summary>
+[Route("authfront/api/v1/systemregister")]
```
Comment (acn-dgopa) [inline src/Authentication/Controllers/SystemRegisterController.cs:13]: We can also discuss about the route organization either in grooming or refinement this week

**Comment by acn-dgopa** [comment]
Looks good and i assume this PR is still draft as you are writing some unit/integration tests

File: src/Authentication/Controllers/SystemRegisterController.cs (hunk 2 above)
Comment (simen-rekkedal) [reply src/Authentication/Controllers/SystemRegisterController.cs:13]: Ok, we need to add Role to the Authorize, but the route remains the same, re discussion in Refinement meeting.

**Pull request merged**